Failure: TestParseReferenceWithTagAndDigest

Currently `master` fails with this error:

```
?       github.com/containers/image     [no test files]
ok      github.com/containers/image/copy        0.052s  coverage: 14.9% of statements
ok      github.com/containers/image/directory   0.052s  coverage: 78.1% of statements
ok      github.com/containers/image/directory/explicitfilepath  0.011s  coverage: 85.7% of statements
ok      github.com/containers/image/docker      4.481s  coverage: 47.1% of statements
ok      github.com/containers/image/docker/daemon       0.008s  coverage: 10.5% of statements
ok      github.com/containers/image/docker/policyconfiguration  0.010s  coverage: 94.7% of statements
--- FAIL: TestParseReferenceWithTagAndDigest (0.00s)
        reference_test.go:248: Error parsing reference: "busybox:latest@sha256:86e0e091d0da6bde2456dbb48306f3956bbeb2eae1b5b9a43045843f69fe4aaa" is not a valid repository/tag
FAIL
coverage: 69.9% of statements
FAIL    github.com/containers/image/docker/reference    0.005s
ok      github.com/containers/image/image       0.039s  coverage: 47.7% of statements
ok      github.com/containers/image/manifest    0.065s  coverage: 90.6% of statements
?       github.com/containers/image/oci [no test files]
ok      github.com/containers/image/oci/layout  0.011s  coverage: 45.4% of statements
ok      github.com/containers/image/openshift   0.005s  coverage: 4.0% of statements
ok      github.com/containers/image/signature   38.552s coverage: 94.2% of statements
ok      github.com/containers/image/storage     0.011s  coverage: 28.8% of statements
ok      github.com/containers/image/transports  0.007s  coverage: 92.9% of statements
?       github.com/containers/image/types       [no test files]
?       github.com/containers/image/version     [no test files]
make: *** [Makefile:18: test] Error 1
make test  11.77s user 2.23s system 33% cpu 41.214 total
```

However this error **only** happens if you update upstream to the latest version (`go get -t`). So I think this is actually some upstream change that we aren't handling properly. Do we want to vendor this code, or otherwise integrate it here? I'm not sure, but something is definitely broken.

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,8 @@ SKOPEO_REPO = projectatomic/skopeo
 SKOPEO_BRANCH = master
 # Set SUDO=sudo to run container integration tests using sudo.
 SUDO =
-BUILDFLAGS = -tags "btrfs_noversion libdm_no_deferred_remove"
+BUILDTAGS   = btrfs_noversion libdm_no_deferred_remove
+BUILDFLAGS := -tags "$(BUILDTAGS)"
 
 all: deps .gitvalidation test validate
 
@@ -30,7 +31,7 @@ test-skopeo:
 		git clone -b $(SKOPEO_BRANCH) https://github.com/$(SKOPEO_REPO) $${skopeo_path} && \
 		rm -rf $${vendor_path} && cp -r . $${vendor_path} && \
 		cd $${skopeo_path} && \
-		make binary-local test-all-local && \
+		make BUILDTAGS="$(BUILDTAGS)" binary-local test-all-local && \
 		$(SUDO) make check && \
 		rm -rf $${skopeo_path}
 

--- a/docker/reference/reference.go
+++ b/docker/reference/reference.go
@@ -6,6 +6,9 @@ import (
 
 	"github.com/pkg/errors"
 
+	// "docker/distribution/digest" requires us to load the algorithms that we
+	// want to use into the binary (it calls .Available).
+	_ "crypto/sha256"
 	"github.com/docker/distribution/digest"
 	distreference "github.com/docker/distribution/reference"
 )
@@ -55,7 +58,7 @@ type Canonical interface {
 func ParseNamed(s string) (Named, error) {
 	named, err := distreference.ParseNamed(s)
 	if err != nil {
-		return nil, errors.Errorf("Error parsing reference: %q is not a valid repository/tag", s)
+		return nil, errors.Wrapf(err, "Error parsing reference: %q is not a valid repository/tag", s)
 	}
 	r, err := WithName(named.Name())
 	if err != nil {

--- a/docker/reference/reference_test.go
+++ b/docker/reference/reference_test.go
@@ -3,6 +3,7 @@ package reference
 import (
 	"testing"
 
+	_ "crypto/sha256"
 	"github.com/docker/distribution/digest"
 )
 

